base/linalg/dense.jl

```diff
@@ -74,10 +71,8 @@ julia> isposdef(A)
 true
 ```
 """
-function isposdef(A::AbstractMatrix{T}) where T
-    S = typeof(sqrt(one(T)))
-    isposdef!(S == T ? copy(A) : convert(AbstractMatrix{S}, A))
-end
+isposdef(A::AbstractMatrix{T}) where {T} =
+    isposdef!(copy_oftype(A, promote_type(typeof(chol(one(T))),Float32)))
```

Comment:
It's from 186287a2a891baf0edf658220b43e3077af00904 but from the comment it seems like it could have been removed in 5e880747d8c0e444863a6676a4ea43c758858999?